controller/internal/controller/deploymentsettings_controller.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	console "github.com/pluralsh/console-client-go"
+	"github.com/pluralsh/console/controller/api/v1alpha1"
+	consoleclient "github.com/pluralsh/console/controller/internal/client"
+	"github.com/pluralsh/console/controller/internal/utils"
+	"github.com/pluralsh/polly/algorithms"
+	"github.com/samber/lo"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	deploymentSettingsName      = "global"
+	deploymentSettingsNamespace = "plrl-deploy-operator"
+)
+
+// DeploymentSettingsReconciler reconciles a DeploymentSettings object
+type DeploymentSettingsReconciler struct {
+	client.Client
+	Scheme        *runtime.Scheme
+	ConsoleClient consoleclient.ConsoleClient
+}
+
+//+kubebuilder:rbac:groups=deployments.plural.sh,resources=deploymentsettings,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=deployments.plural.sh,resources=deploymentsettings/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=deployments.plural.sh,resources=deploymentsettings/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.3/pkg/reconcile
+func (r *DeploymentSettingsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	logger := log.FromContext(ctx)
+
+	settings := &v1alpha1.DeploymentSettings{}
+	if err := r.Get(ctx, req.NamespacedName, settings); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// make sure there is only one CRD object with the `global` name and the agent namespace
+	if settings.Name != deploymentSettingsName || settings.Namespace != deploymentSettingsNamespace {
```

Comment:
@michaeljguarino I've added such a check to make sure we have only one DeploumentSettings CRD object
with the `global` name and the agent namespace `plrl-deploy-operator`.

Reply:
Ok cool that's probably wise